src/goose/synopsis/bash.py

```diff
@@ -0,0 +1,63 @@
+import os
+from pathlib import Path
+
+from rich.markdown import Markdown
+from rich.rule import Rule
+from goose.notifier import Notifier
+from goose.view import ExchangeView
+from goose.toolkit.utils import RULEPREFIX, RULESTYLE
+from goose.synopsis.system import system
+from goose.utils.shell import shell
+
+
+class Bash:
+    def __init__(self, notifier: Notifier, exchange_view: ExchangeView) -> None:
+        self.notifier = notifier
+        self.exchange_view = exchange_view
+
+        # Command dispatch dictionary
+        self.command_dispatch = {
+            "source": self._source,
+            "shell": self._shell,
+            "change_dir": self._change_dir,
+        }
+
+    def logshell(self, command: str, title: str = "shell") -> None:
+        self.notifier.log("")
+        self.notifier.log(
+            Rule(RULEPREFIX + f"{title} | [dim magenta]{os.path.abspath(system.cwd)}[/]", style=RULESTYLE, align="left")
```

Comment:
:nit: since we already use DI in a few places, we should do it here as well (e.g. `self.system = system`